common-system-test/src/main/java/com/teamscale/test/commons/TeamscaleMockServer.java

```diff
@@ -52,20 +52,11 @@ public class TeamscaleMockServer {
 	public final Set<ClusteredTestDetails> availableTests = new HashSet<>();
 	private final Path tempDir = Files.createTempDirectory("TeamscaleMockServer");
 	private final Service service;
-	private final List<String> impactedTests;
+	private List<String> impactedTests;
 
-	public TeamscaleMockServer(int port, boolean expectNoInteraction, String...impactedTests) throws IOException {
-		this.impactedTests = Arrays.asList(impactedTests);
+	public TeamscaleMockServer(int port) throws IOException {
 		service = Service.ignite();
 		service.port(port);
-		if (expectNoInteraction) {
-			service.post("", (request, response) -> { throw new IllegalStateException("No POST to the mock server expected!"); });
-			service.put("", (request, response) -> { throw new IllegalStateException("No PUT to the mock server expected!"); });
-		} else {
-			service.post("api/v5.9.0/projects/:projectName/external-analysis/session/auto-create/report",
-					this::handleReport);
-			service.put("api/v8.0.0/projects/:projectName/impacted-tests", this::handleImpactedTests);
-		}
```

Comment:
At first I wanted to add a system test and then thought that adding yet another boolean flag or similar here would not be a very clean solution, so I refactored the class to a builder pattern like structure which allows to easily add additional cases like "return 403 for all requests" etc. WDYT?